### PR TITLE
23 new donated item form

### PR DESCRIPTION
**Pull request by truffer11**
**Fixes Issue #23 **

### What was changed?

- Added a new form for newly donated items in client-app/src/components.

### Why was it changed?

- Allows for a user friendly and efficient way to add newly donated items to the database.
- - Note: This code is not yet connected to the database, however it will be connected in the future.
- - The form is way easier to understand and fill out compared to directly entering this data into the database.

### How was it changed?

- There is a new file, NewItemForm.tsx where the code should live eventually.
- For now, just so I wasn't creating the form blind, the code is in DonorForm.tsx as well. The original code for DonorForm.tsx is commented out below. This will all be fixed as soon as I do not need to see the New Item Form.
- On the foundational level, this code is similar to the code used in the Donor Form. I decided to do this to keep consistency among the forms, keeping the same style and color formatting. 
-  - The rest of the code has been changed to allow for different fields, dropdowns, dates, and file uploads.
-  - As it relates to error handling, there have been a few modifications, including field specific errors and new item exclusive checks.
- It should be important to note that this code is not yet set up with the database, however that will be done at a later time.
- Images are stored in an array as base64 strings instead of their file names.

### Screenshots that show the changes (if applicable):

- **Before:**
  - *Crickets*
- **After:**
  - image_2024-09-30_153213489
  - Screenshot 2024-09-30 153103

### CODE REVIEW

File: client-app/src/Components/DonorForm.tsx [hunk 1]
```diff
@@ -2,6 +2,294 @@ import React, { useState, ChangeEvent, FormEvent } from 'react';
 import axios from 'axios';
 import '../css/DonorForm.css';
 
+interface FormData {
+    itemType: string;
+    currentStatus: string;
+    donorEmail: string;
+    program: string;
+    imageUpload: string[];
```
Comment (miabu-pashh) [inline client-app/src/Components/DonorForm.tsx:10]: Image Upload is stored as a string array[], but it should have been stored as a file upload.

File: client-app/src/Components/DonorForm.tsx [hunk 2]
```diff
@@ -2,6 +2,294 @@ import React, { useState, ChangeEvent, FormEvent } from 'react';
...
+        currentStatus: 'Received',
+        donorEmail: '',
+        program: '',
+        imageUpload: [],
+        dateDonated: '',
+    });
+
+    const itemTypeOptions = [
+        {value: 'bicycle', label: 'Bicycle'},
+        {value: 'computer', label: 'Computer'},
```
Comment (miabu-pashh) [inline client-app/src/Components/DonorForm.tsx:30]: More item types may be added for the next sprint as the user have more options to donate.

File: client-app/src/Components/DonorForm.tsx [hunk 3]
```diff
@@ -2,6 +2,294 @@ import React, { useState, ChangeEvent, FormEvent } from 'react';
...
+
+    const programOptions = [
+        {value: 'youthProgram', label: 'Youth Program'},
+        {value: 'retailSales', label: 'Retail Sales'},
+        {value: 'recycle', label: 'Recycle'},
+        {value: 'earnABicycle', label: 'Earn-a-bicycle'},
+        {value: 'earnAComputer', label: 'Earn-a-computer'},
+    ]
+
+    const convertToBase64 = (file: File): Promise<string> => {
```
Comment (miabu-pashh) [inline client-app/src/Components/DonorForm.tsx:47]: Image conversion to base64 function looks fine as it converts the file to textual format.

File: client-app/src/Components/DonorForm.tsx [hunk 4]
```diff
@@ -2,6 +2,294 @@ import React, { useState, ChangeEvent, FormEvent } from 'react';
...
+    };
+
+    // Generalized validation function to reduce code repetition
+    const validateField = (name: string, value: any) => {
+        const requiredFields = [
+            'itemType',
+            'currentStatus',
+            'donorEmail',
+            'program',
+            'imageUpload',
```
Comment (miabu-pashh) [inline client-app/src/Components/DonorForm.tsx:89]: The image upload option should be added for the backend database model as well.

File: client-app/src/Components/NewItemForm.tsx [hunk 5]
```diff
@@ -0,0 +1,285 @@
...
+                    >
+                        Refresh
+                    </button>
+                </div>
+            </form>
+        </div>
+    );
+};
+
+export default NewItemForm;
```
Comment (miabu-pashh) [inline client-app/src/Components/NewItemForm.tsx:285]: This is same as donor form, this NewItemForm file should be imported in the home,  or else both donor form and donated item from should be integrated. Separate routes should be created.

**Review by miabu-pashh — COMMENTED** [review]
Overall, two separate tsx files should be written for donor form and donated item form, and image upload option should be implemented in the backend option as well. as multiple images are being uploaded, we need to use an array of objects. The file handling function looks fine and form validators with error messages are neat. The donor should be able to have more options to donate.

File: client-app/src/Components/NewItemForm.tsx [hunk 6]
```diff
@@ -0,0 +1,286 @@
+import React, { useState, ChangeEvent, FormEvent } from 'react';
```
Comment (Anjali0407-git) [inline client-app/src/Components/NewItemForm.tsx:1]: @truffer11 rename the file to DonatedItemForm.tsx

**Pull request merged**